### ISSUE TITLE
sample bandcenter, bandwidth and fwhm for mega-run

### ISSUE THREAD

**Issue by zonca** [hpc4cmb/toast#167]
@tskisner  when you generate the focalplane we need also to sample bandpass and beam from gaussian.

Here is how I set those parameters in `toast_fake_focalplane.py`:

https://github.com/zonca/toast/blob/c9644878b0b95bf295109f9da61d2b1ef116622c/pipelines/toast_fake_focalplane.py#L89

However here are all equal, do you already plan to write a dedicated script to generate the focalplane for the mega-run?
Let me know if you prefer to add this feature yourself or if you prefer I do it.

**Comment by keskitalo** [comment]
Hey @zonca. How do you plan on propagating the information from the focalplane into the PySM operator? As a separate container or are you perhaps inserting the information into the observation dictionary? I will need the bandpasses in the atmospheric operator so we better use the same numbers.

**Comment by zonca** [comment]
I pass the focalplanes as input to the operator, then the operator looks into the focalplanes dictionaries for the bandpass information, see the link above for the parameter keys

**Comment by keskitalo** [comment]
The ground simulation pipeline supports multiple telescopes and focalplanes. Making sure you apply the right focalplane to the right observations may get complicated. I think it would be best to link the focalplane to the observation:
```
for obs in data.obs:
    tod = obs['tod']
    focalplane = obs['focalplane']
```
I can quickly adjust the ground simulation script to do the linking when building the observations.

What do you think?

**Comment by zonca** [comment]
that looks too nested and also a huge duplication of the data structure.
the only requirement now is that a detector name is unique. Is that a problem?

**Comment by keskitalo** [comment]
We may not have control over the focal planes we are given. The name duplication is possible. I think an observation should be able to provide the focal plane when needed. It is only a reference, so the data itself are not duplicated unless the pipeline enforces it.

**Comment by zonca** [comment]
can you modify this after I merge my branch?

**Comment by keskitalo** [comment]
Sure!

**Comment by tskisner** [comment]
Hi @zonca and @keskitalo, for the immediate case of the next 2 days, anything that works is fine.  I think that longer term we probably need some more formal definition in toast of what we mean by a "focalplane".   Currently, the TOD class is where the detector names and quaternion offsets "live", not in the observation dictionary.  This class is also where the data distribution along the time and detector axes is done.

At the moment, we completely control all detector names and how the focalplanes are created, so we just need to be consistent.  However, in the future we should definitely generalize the focalplane information we are passing to the TOD class or change the TOD and observation concepts somehow.

**Comment by zonca** [comment]
@tskisner as per our discussion yesterday, you'll start implementing the script to generate the focalplane, then if you want I can help adding the bandpass and beam sampling.

**Comment by zonca** [comment]
@keskitalo @tskisner is the script you used to generate the focalplane available on Github?

**Comment by keskitalo** [comment]
It is not. @tskisner do you have a place in mind for little helper scripts that we'll have use for preparing for future runs?

**Comment by zonca** [comment]
@tskisner @keskitalo any update on this?

**Comment by keskitalo** [comment]
Addressed in #171. We should open another issue about where to store the little helper scripts.
